tests/load/pipeline/test_pipelines.py

```diff
@@ -997,3 +997,158 @@ def events_timezone_unset():
+            {"id": 300, "name": "bobby"},  # the lag will be applied here
+        ]
+
+        # Updated entries in second call
+        second_run_events = [
+            {"id": 100, "name": "updated"},
+            {"id": 200, "name": "updated"},
+            {"id": 300, "name": "updated"},
+            {"id": 400, "name": "mark"},  # the lag will be applied here
+        ]
+
+        third_run_events = [
+            {"id": 100, "name": "updated"},
+            {"id": 200, "name": "updated"},
+            {"id": 300, "name": "updated"},
+            {"id": 400, "name": "updated"},
+            {"id": 500, "name": "nicolas"},
+        ]
+
+        if is_second_run:
+            yield from second_run_events
+        elif is_third_run:
+            yield from third_run_events
+        else:
+            yield from initial_entries
+
+    pipeline.run(items_resource)
+    is_second_run = True
+    pipeline.run(items_resource)
+    is_second_run = False
+    is_third_run = True
+    pipeline.run(items_resource)
+
+    expected_results = {
+        100: 9,
+        10: 7,
+        1: 7,
+        0: 5,
+        -1: 5,
+    }
+
+    # Validate final dataset
+    with pipeline.sql_client() as sql_client:
+        assert (
+            sql_client.execute_sql(f"SELECT count(*) FROM {name}")[0][0]
+            == expected_results[int(lag)]
+        )
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(
+        default_staging_configs=True, default_sql_configs=True, subset=["postgres", "duckdb"]
+    ),
+    ids=lambda x: x.name,
+)
+@pytest.mark.parametrize("lag", [3601, 3600, 60, -1])
+def test_pipeline_resource_incremental_datetime_lag(
+    destination_config: DestinationTestConfiguration, lag: float
+) -> None:
+    """
+    Test incremental lag behavior for datetime data while using `id` as the primary key.
+
+    We validate that the final dataset reflects the correct data updates, taking the lag into account,
+    and ensure deduplication is disabled when lag is set.
+    """
+
+    dataset_name = f"{destination_config.destination_name}{uniq_id()}"
+    pipeline = destination_config.setup_pipeline("pipeline", dataset_name=dataset_name)
+
+    name = "events"
+    is_second_run = False
+
+    @dlt.resource(name=name, primary_key="id", write_disposition="merge")
+    def events_resource(_=dlt.sources.incremental("created_at", lag=lag)):
+        nonlocal is_second_run
+
+        # Initial entries
+        initial_entries = [
+            {"id": 1, "created_at": "2023-03-03T01:00:00Z", "event": "1"},
+            {"id": 2, "created_at": "2023-03-03T01:00:01Z", "event": "2"},
+            {
+                "id": 3,
+                "created_at": "2023-03-03T02:00:01Z",
+                "event": "3",
+            },  # the lag will be applied here
+        ]
+
+        # Updated entries in second call
+        updated_events = [
+            {"id": 1, "created_at": "2023-03-03T01:00:00Z", "event": "updated"},
```

Comment:
I'd prefer if you name it updated_1, updated_2 etc. so it is clearly visible which updated got it. way easier to verify the test